### PR TITLE
fix(route53-targets): A/AAAA Alias Record to ELB cannot resolve IPv6 addresses

### LINKED ISSUE #6271: Route53 A/AAAA Alias Record to ELB does not use "dualstack" prefix
When creating an ALIAS A/AAAA record in Route53 pointing to an ELB on the Web Console, the name will always have a "dualstack" prefix, as also suggested by the AWS documentation: https://aws.amazon.com/de/premiumsupport/knowledge-center/alias-resource-record-set-route53-cli/
> If you're pointing to an Elastic Load Balancer, always include dualstack in the value for the DNSName key-value pair of the JSON file.

This ensures that both IPv4 _and_ IPv6 addresses are solved.

However, a CDK ARecord as well as a AaaaRecord with a LoadBalancerTarget will not add this "dualstack" prefix.

### Reproduction Steps
```typescript
import * as route53 from '@aws-cdk/aws-route53';
import * as route53targets from '@aws-cdk/aws-route53-targets';
...
const elbZoneAAAARecord = new route53.AaaaRecord(this, 'ElbAaaaRecord', {
  zone: zone,
  target: route53.RecordTarget.fromAlias(new route53targets.LoadBalancerTarget(elb)),
});
```
When looking at Route53 record in the the Web Console, the record will not have the "dualstack" prefix.

### Environment

  - **CLI Version      :** 1.23.0 (build 01f326e)
  - **Framework Version:** 1.23.0 (build 01f326e)
  - **OS               :** Windows 10 x64
  - **Language         :** typescript

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by BryanPan342**
**[ISSUE]**
Route53 A/AAAA Alias Record created by CDK didn't add prefix 'dualstack' to the dnsName of ElasticLoadBalancer as suggested in [documentation](https://aws.amazon.com/de/premiumsupport/knowledge-center/alias-resource-record-set-route53-cli/), resulting in inability to resolve IPv6 records (outcome consistent with these [docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-internet-facing-load-balancers.html)).

**[APPROACH]**
Joined prefix: _dualstack_ to DNS of ELB targets to resolve IPv6 records. 

Example:
> If ELB DNS name is `ALB-xxxxxxxx.us-west-2.elb.amazonaws.com`,
> Route53 A/AAAA Alias Record will create Alias Target `dualstack.ALB-xxxxxxxx.us-west-2.elb.amazonaws.com`

**[NOTE]** 
Ignoring the case of using `ApiGatewayDomain(domainName)` because the prefix of 'dualstack' should be added in the `domainName` parameter.

```
new route53.ARecord(this, 'AliasRecord', {
  zone,
  target: route53.RecordTarget.fromAlias(new alias.ApiGatewayDomain(domainName)),
});
```

If creating an ApiGatewayDomain for ELB, endpoint for ELB with prefix 'dualstack' should be declared there, not within route53 package.

Fixes #6271 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-route53-targets/lib/classic-load-balancer-target.ts [hunk 1]
```diff
@@ -9,9 +10,11 @@ export class ClassicLoadBalancerTarget implements route53.IAliasRecordTarget {
   }
 
   public bind(_record: route53.IRecordSet): route53.AliasRecordTargetConfig {
+    const delimiter: string = '.';
+    const prefix: string[] = ['dualstack'];
     return {
       hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneNameId,
-      dnsName: this.loadBalancer.loadBalancerDnsName,
+      dnsName: cdk.Fn.join(delimiter, prefix.concat([this.loadBalancer.loadBalancerDnsName])),
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53-targets/lib/classic-load-balancer-target.ts:17]:
```suggestion
      dnsName: cdk.Fn.join('.', ['dualstack', this.loadBalancer.loadBalancerDnsName]),
```

File: packages/@aws-cdk/aws-route53-targets/lib/load-balancer-target.ts [hunk 2]
```diff
@@ -9,9 +10,11 @@ export class LoadBalancerTarget implements route53.IAliasRecordTarget {
   }
 
   public bind(_record: route53.IRecordSet): route53.AliasRecordTargetConfig {
+    const delimiter: string = '.';
+    const prefix: string[] = ['dualstack'];
     return {
       hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
-      dnsName: this.loadBalancer.loadBalancerDnsName,
+      dnsName: cdk.Fn.join(delimiter, prefix.concat([this.loadBalancer.loadBalancerDnsName])),
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-route53-targets/lib/load-balancer-target.ts:17]: same changes as suggested in classic

File: yarn.lock [hunk 3]
```diff
@@ -1887,6 +1887,11 @@ anymatch@^3.0.3:
     normalize-path "^3.0.0"
     picomatch "^2.0.4"
 
+app-root-path@^2.2.1:
```
[the remark this answers is not in the document]
Comment (BryanPan342) [reply yarn.lock:1890]: oh my gosh, I knew it was a dependency bug but didn't think it was that easy 😭

File: packages/@aws-cdk/aws-route53-targets/lib/load-balancer-target.ts [hunk 4]
```diff
@@ -11,7 +12,7 @@ export class LoadBalancerTarget implements route53.IAliasRecordTarget {
   public bind(_record: route53.IRecordSet): route53.AliasRecordTargetConfig {
     return {
       hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
-      dnsName: this.loadBalancer.loadBalancerDnsName,
+      dnsName: cdk.Fn.join('.', ['dualstack', this.loadBalancer.loadBalancerDnsName]),
```
Comment (jogold) [inline packages/@aws-cdk/aws-route53-targets/lib/load-balancer-target.ts:15]: Why not ``dualstack.${this.loadBalancer.loadBalancerDnsName}`` and let `core` handle the case when a `Fn.join` is effectively needed.

Comment (shivlaks) [reply packages/@aws-cdk/aws-route53-targets/lib/load-balancer-target.ts:15]: even better, let's go with that!

**Comment by BryanPan342** [comment]
> nice job getting this one started!!!
> 
> I have a couple of suggestions/questions:
> 
> 1. Confirm this is not a breaking change
> 
> * We are updating an existing property here and we want to make sure we don't break customers as this is a stable module. From the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html#aws-properties-route53-recordset-properties), it looks like update to `AliasTarget` requires no interruption and so it should be safe to introduce. Worth verifying that the documentation is accurate.
> 
> 1. Commit Message & Body
> 
> * In your commit description, outline the root cause of the issue and describe the approach taken to fix, any alternatives, considerations, caveats, gotchas, etc. You don't need the code snippets in the commit body
> * change the commit message to describe the issue. The reason for this is because this will show up in the [release notes](https://github.com/aws/aws-cdk/releases/tag/v1.47.0) as bug fixes. It's easier for users to read it when the message describes the bug that was fixed. In this case it might be something like ` A/AAAA Alias Record to ELB cannot resolve IPv4 and IPv6 addresses`
> 
> 1. Update the README.
> 
> * It's a little plain right now, but let's start improving the README with changes that are in scope for this fix. That might simply be adding some documentation links and describing the behaviour since that's changing now.
> 
> I believe you also have some tests in other modules that use the ELB target in their tests need to be updated

no longer a breaking change ✅
updated commit message ✅
updated readme ✅

**Pull request merged**